doc/src/guides/solving/solve-diophantine-equation.md

```diff
@@ -0,0 +1,207 @@
+# Solve a Diophantine Equation Algebraically
+
+Use SymPy to solve a Diophantine equation (find integer solutions to a
+polynomial equation) algebraically, returning a parameterized general solution
+if possible. For example, solving the [Pythagorean
+theorem](https://en.wikipedia.org/wiki/Pythagorean_theorem) $a^2 + b^2 = c^2$
+yields $(a=2pq, b=p^2-q^2, c=p^2-q^2)$.
+
+## Alternatives to Consider
+
+There are few alternatives for finding a parameterized general solution a
+Diophantine equation.
+- Numerical alternatives:
+    - [Sage's EllipticCurve
+  command](https://doc.sagemath.org/html/en/constructions/elliptic_curves.html)
+  may be able to find a set of relative numerical values for each variable
+    - You can test explicit integer values, for example using a nested for loop
+  of ranges of values. This is inefficient, but fine if you are only interested
+  in solutions that are relatively small.
+- {func}`~.solve` simply solves for one variable in terms of the others. For
```

Comment:
solve also treats the variables as real or complex numbers, which produces a different type of solution.